packages/thirdweb/src/transaction/transaction-store.ts

```diff
@@ -0,0 +1,57 @@
+import { type Store, createStore } from "../reactive/store.js";
+import type { Address } from "../utils/address.js";
+import type { Hex } from "../utils/encoding/hex.js";
+
+type StoredTransaction = {
+  transactionHash: Hex;
+  chainId: number;
+};
+
+const transactionsByAddress = new Map<Address, Store<StoredTransaction[]>>();
+
+/**
+ * Retrieve transactions for the given account for the current session. Sessions are stored in memory and are not persisted.
+ * @param account {Account} - The account to retrieve transactions for.
+ * @returns {StoredTransaction[]} An array of stored pending and confirmed transactions.
+ */
+export async function getSessionTransactions(
+  address: Address,
+): Promise<StoredTransaction[]> {
+  const tranasctionStore = transactionsByAddress.get(address);
```

Comment:
probably instead simply use `getTransactionStore()`, there is no harm creating an empty store in this case

Reply:
.. and that way you have 1 clear path to how to interact with the map